src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -562,6 +586,8 @@
         this)
       (a/start [this]
         (run-lifecycle-fns app-context s/start "start" ordered-services)
+        (let [restart-file (get-in-config (a/get-service this :ConfigService) [:global :restart-file])]
```

Comment:
If you use `when-let` instead of `let` here, it will only fire when there is a restart-file defined.